### DISCUSSION TITLE
Ansteuerung 36V Akku über PV 1 eingang

### DISCUSSION THREAD

**Discussion by Padrino1974** [svenerbe/ecoflow_dynamic_power_adjustment#34] [General]
Erst einmal besten Dank für Deine Arbeit und die Tolle Anleitung. Funktioniert dieses Skript auch wenn man keinen Ecoflow  Akku installiert hat. Ich nutze den Powerstream nur dazu um über 36 V Akkus Strom bzw. Grundlast einzuspeisen.  Vorzugsweise PV1.

Wie müsste ggf. das Skript angepasst werden.

ich nutze aktuell folgenden Skript:

alias: ecoflow-powerstream-dynamische-last
description: ecoflow-powerstream-dynamische-last
trigger:
  - platform: time_pattern
    seconds: /3
condition: []
action:
  - service: pyscript.set_ef_powerstream_custom_load_power
    data:
      SerialNumber: "HW51xxxxxxx"
      TotalPower: >-
        {{ (states('sensor.bitshake_smartmeterreader_lk13be_power') | float) +
        (states('sensor.powerstream_inverter_output_watts') | float) -2 | round 
        }}
      Automation: false
mode: single

**Comment by Padrino1974** [comment]
Ich habe das Skript noch einmal angepasst aber eine Einspeisung von PV 1 findet bei anliegenden 42 v nicht statt

action: pyscript.set_ef_powerstream_custom_load_power
data:
  SerialNumber: HW51xxxxxx
  TotalPower: >-
    {{ (states('sensor.bitshake_smartmeterreader_lk13be_power') | float) +
    (states('sensor.powerstream_inverter_output_watts') | float) - 2 | round  }}
  Automation: false

**Comment by Just-Zuul** [comment]
Probier mal:

TotalPower: >
{{ (states('sensor.bitshake_smartmeterreader_lk13be_power') | float) }}

besser noch:

TotalPower: >
{{ (states('sensor.bitshake_smartmeterreader_lk13be_power') | float) -20 }}

sonst verschenkst Du mehr als es was bringt....

der Teil mit :
 + (states('sensor.powerstream_inverter_output_watts') | float) - 2 | round }}

ist unsinnig

**Comment by Padrino1974** [comment]
Danke für Deine Hilfe, ich habe es eingegeben, nur zum Verständnis ist es bei diesem Befahl egal ob die Einspeisung über PV Module oder Akku am PV Eingang ankommt? Könnte man zwischen PV1 mit angeschlossenen PV Modulen und PV2 angeschlossenen Akkus unterscheiden?

**Reply by Just-Zuul** [reply]
Ich glaube, Du verstehst nicht, was das Script macht und wie es funktioniert.

Die Automatisation übergibt einen Wert an das pyscript. Der Wert stammt in dem Fall von Deinem Zählersensor (die momentane Wattleistung die Du aus dem Netz beziehst * siehe auch unten). Du kannst jeden X-beliebigen Wert, auch variable Werte (input_number-Feld) nehmen oder sogar einen festen Wert reinschreiben - ist dem Script egal. Die Automatisation ist nur dafür da, den Wert in bestimmten Abständen an das Pyscript zu senden (ideal sind so alle 4 Sekunden).

Das pyscript nimmt den Wert den es erhält und schickt den Wert an die API und die schickt den Wert an Deinen Powerstream (über die EcorFlow Cloud so als wenn Du den Regler in Deiner App bewegst) der Wert landet im Feld der Grundbedarfseinstellung und der Inverter gibt dann diesen Wert bis zur nächsten Änderung aus.

Woher Dein Inverter den Strom nimmt ist dem Script egal, das weiß es ja nicht - es setzt nur den Wert in das Feld für die Grundeinspeisung (other_loads) für den Inverter.

Ob es eine gute Idee ist, eine Batterie an einen MPPT-Solareingang zu hängen, kann ich nicht beantworten. Ob das der Powerstrem mitmacht und wie lange oder zuverlässig auch nicht. Das ist nicht mein Fachbereich.

Prinzipiell beeinflusst das Script nur den Wert in dem Feld sensor.{invertername}_other_loads - mehr nicht. Das macht es egal ob der Inverter Strom bekommt oder nicht (nur das der Inverter ohne Strom gar nichts macht). So, als wenn du den Regler in Deiner App bewegst....

Was im Feld "Grundeinspeisung" drin steht ist das, was der Inverter dann abgibt wenn er den Strom auch hat / bekommt. Daher ist die Frage ob man Strom über PV1 oder 2 an den Inverter gibt für diese Funktion irrelevant. Das, was der Inverter bekommt, kann er auch abgeben und abgeben tut er maximal das, was man einstellt.

Ich hoffe Deine Frage damit besser beantwortet zu haben. Viel Erfolg weiterhin.


* Ach so - vergessen - Du musst natürlich den richtigen Sensor Deines Zählers nehmen !!

Du hast angegeben:   sensor.bitshake_smartmeterreader_lk13be_power

Da ich auch einen lk13be habe, wundert mich der Name des Sensors.
Bei mir heißt der Sensor am Ende: "Power_curr" für die momentane Gesamtleistung in Watt. Es gibt da mehrere (4) "Power" - Du musst natürlich den nehmen, der Deine Wattage komplett abbildet wenn Du die Gesamtlast als Wert ausgleichen willst.
